src/shared/pam-util.c

```diff
@@ -50,10 +44,8 @@ int pam_acquire_bus_connection(pam_handle_t *handle, sd_bus **ret) {
         }
 
         r = sd_bus_open_system(&bus);
-        if (r < 0) {
-                pam_syslog(handle, LOG_ERR, "Failed to connect to system bus: %s", strerror_safe(r));
-                return PAM_SERVICE_ERR;
-        }
+        if (r < 0)
+                return pam_syslog_errno(handle, LOG_ERR, r, "Failed to connect to system bus: %m");
```

Comment:
hmm, this changes the returned error PAM_SERVICE_ERR → PAM_BUF_ERR, afaics. I am not sure we should do that. The latter is an OOM-like error iirc, i.e. probably should result in complete failure, while the former might be reason to repeat.

I'd really stick to the returned error here...

Reply:
Yeah, @yuwata also noticed that. I changed the code to return PAM_BUF_ERR for ENOMEM and PAM_SERVICE_ERR in other cases.